src/Package_Command.php

```diff
@@ -1358,4 +1388,55 @@ static function () use (
 	private function is_composer_v2() {
 		return version_compare( Composer::getVersion(), '2.0.0', '>=' );
 	}
+
+	/**
+	 * Try to retrieve default branch via GitHub API.
+	 *
+	 * @param string $package_name GitHub package name to retrieve the default branch from.
+	 * @param bool   $insecure     Optional. Whether to insecurely retry downloads that failed TLS handshake. Defaults
+	 *                             to false.
+	 * @return string Default branch, or 'master' if it could not be retrieved.
+	 */
+	private function get_github_default_branch( $package_name, $insecure = false )
+	{
+		$github_token    = getenv( 'GITHUB_TOKEN' ); // Use GITHUB_TOKEN if available to avoid authorization failures or rate-limiting.
+		$headers         = $github_token ? [ 'Authorization' => 'token ' . $github_token ] : [];
+		$options         = [ 'insecure' => $insecure ];
+
+		$github_api_repo_url = "https://api.github.com/repos/{$package_name}";
+		$response = Utils\http_request( 'GET', $github_api_repo_url, null /*data*/, $headers, $options );
+		if ( 20 !== (int) substr( $response->status_code, 0, 2 ) ) {
+			WP_CLI::warning(
+				sprintf(
+					"Couldn't fetch default branch for package '%s' (HTTP code %d). Presuming default branch is 'master'.",
+					$package_name,
+					$response->status_code
+				)
+			);
+			return 'master';
```

Comment:
AFAIK nowadays the default branch for new repositories on GitHub is `main`, but I suppose that it might still be `master` for older ones that haven't switched.

Reply:
Yes, I'm keeping the `master` for now as GitHub and Composer seem to keep aliases around for BC. This can still be changed at a later date.